File: Capítulo_13/Algoritmos de busca/Busca sequencial/Código em Python/sequentialSearch.py
# Função defaultEquals
def default_equals(a, b):
    return a == b

# Variável DOES_NOT_EXIST
DOES_NOT_EXIST = -1

# Algoritmo sequential search
def sequential_search(array, value, equalsFn = default_equals):
    for i in range(len(array)):
        if equalsFn(value, array[i]):
            return i
    
    return DOES_NOT_EXIST

# Testando algortimo sequential_search
array = [5, 4, 3, 2, 1]

print(sequential_search(array, 3)) # 2

File: Capítulo_13/Algoritmos de busca/Busca sequencial/sequentialSearch.js
import { defaultEquals } from "../../Models/util.js";

const DOES_NOT_EXIST = -1;

// Algoritmo sequential search
function sequentialSearch(array, value, equalsFn = defaultEquals) {
    for (let i = 0;i < array.length; i++) {
        if (equalsFn(value, array[i])) {
            return i;
        }
    }
    return DOES_NOT_EXIST;
}

// Testando algortimo sequentialSearch
let array = [5, 4, 3, 2, 1];

console.log(sequentialSearch(array, 3)) // 2
